homeassistant/components/emonitor/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for SiteSage Emonitor integration."""
+import logging
+
+from aioemonitor import Emonitor
+import aiohttp
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.components.dhcp import IP_ADDRESS, MAC_ADDRESS
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.device_registry import format_mac
+
+from . import name_short_mac
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def fetch_mac_and_title(hass: core.HomeAssistant, host):
+    """Validate the user input allows us to connect."""
+    session = aiohttp_client.async_get_clientsession(hass)
+    emonitor = Emonitor(host, session)
+    status = await emonitor.async_get_status()
+    mac_address = status.network.mac_address
+    return {"title": name_short_mac(mac_address[-6:]), "mac_address": mac_address}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SiteSage Emonitor."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize Emonitor ConfigFlow."""
+        self.discovered_ip = None
+        self.discovered_info = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await fetch_mac_and_title(self.hass, user_input[CONF_HOST])
+            except aiohttp.ClientError:
+                errors[CONF_HOST] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(
+                    format_mac(info["mac_address"]), raise_on_progress=False
+                )
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
We should abort for already configured entry with same unique id here.

Reply:
done in https://github.com/home-assistant/core/pull/48700